### PR TITLE
[I] refact service usage to avoid using android services

### PR DESCRIPTION
**Pull request by StevenWatremez**

### CODE REVIEW

**Review by anthonyGuiguen — APPROVED** [review]

File: android-library/src/main/java/com/zetapush/library/SmartClient.java [hunk 1]
```diff
@@ -15,20 +15,20 @@
     private String deployId_default_weak = "weak_0";
 
     // Constructor
-    public SmartClient(Activity activity) {
-        super(activity);
+    public SmartClient(Context context) {
```
Comment (jgoas) [inline android-library/src/main/java/com/zetapush/library/SmartClient.java:18]: Please annotate this context with @NonNull

File: android-library/src/main/java/com/zetapush/library/SmartClient.java [hunk 2]
```diff
@@ -15,20 +15,20 @@
...
+        super(context);
     }
 
-    public SmartClient(Activity activity, StorageTokenInterface storageTokenHandler) {
-        super(activity, storageTokenHandler);
+    public SmartClient(Context context, StorageTokenInterface storageTokenHandler) {
```
Comment (jgoas) [inline android-library/src/main/java/com/zetapush/library/SmartClient.java:22]: Please annotate this context with @NonNull

File: android-library/src/main/java/com/zetapush/library/SmartClient.java [hunk 3]
```diff
@@ -15,20 +15,20 @@
...
+        super(context, storageTokenHandler);
     }
 
-    public SmartClient(Activity activity, StorageCredentialsInterface storageCredentialHandler) {
-        super(activity, storageCredentialHandler);
+    public SmartClient(Context context, StorageCredentialsInterface storageCredentialHandler) {
```
Comment (jgoas) [inline android-library/src/main/java/com/zetapush/library/SmartClient.java:26]: Please annotate this context with @NonNull

File: android-library/src/main/java/com/zetapush/library/WeakClient.java [hunk 4]
```diff
@@ -11,26 +11,26 @@
     private String resource_default = "android";
     private String deployId_default = "weak_0";
 
-    public WeakClient(Activity activity) {
-        super(activity);
+    public WeakClient(Context context) {
```
Comment (jgoas) [inline android-library/src/main/java/com/zetapush/library/WeakClient.java:14]: Please annotate this context with @NonNull

File: android-library/src/main/java/com/zetapush/library/WeakClient.java [hunk 5]
```diff
@@ -11,26 +11,26 @@
...
+        super(context);
     }
 
-    public WeakClient(Activity activity, StorageTokenInterface storageTokenHandler) {
-        super(activity, storageTokenHandler);
+    public WeakClient(Context context, StorageTokenInterface storageTokenHandler) {
```
Comment (jgoas) [inline android-library/src/main/java/com/zetapush/library/WeakClient.java:18]: Please annotate this context with @NonNull

File: android-library/src/main/java/com/zetapush/library/WeakClient.java [hunk 6]
```diff
@@ -11,26 +11,26 @@
...
+        super(context, storageTokenHandler);
     }
 
-    public WeakClient(Activity activity, StorageCredentialsInterface storageCredentialHandler) {
-        super(activity, storageCredentialHandler);
+    public WeakClient(Context context, StorageCredentialsInterface storageCredentialHandler) {
```
Comment (jgoas) [inline android-library/src/main/java/com/zetapush/library/WeakClient.java:22]: Please annotate this context with @NonNull

File: android-library/src/main/java/com/zetapush/library/ZetaPushService.java [hunk 7]
```diff
@@ -185,71 +178,53 @@ public String getUserKey() {
...
-            sendBroadcast(i);
+            Log.d("ZP", "successfulHandshake -> map = " + map);
+            storageTokenHandler.saveToken((String) map.get("token"));
         }
 
         @Override
-        public void failedHandshake(Map<String, Object> map) {
-
+        public void failedHandshake(String s, ZetaApiError zetaApiError) {
+            Log.d("ZP", "failedHandshake -> message = " + s + ", error = " + zetaApiError);
```
Comment (jgoas) [inline android-library/src/main/java/com/zetapush/library/ZetaPushService.java:203]: Create a constante for your TAG ?

Comment (jgoas) [reply android-library/src/main/java/com/zetapush/library/ZetaPushService.java:203]: A common constant for all the library

Comment (StevenWatremez) [reply android-library/src/main/java/com/zetapush/library/ZetaPushService.java:203]: I create an issue to do this later

**Pull request merged**